src/support/file.cpp

```diff
@@ -26,7 +26,8 @@ T wasm::read_file(const std::string &filename, Flags::BinaryOption binary, Flags
   if (debug == Flags::Debug) std::cerr << "Loading '" << filename << "'..." << std::endl;
   std::ifstream infile;
   std::ios_base::openmode flags = std::ifstream::in;
-  if (binary == Flags::Binary) flags |= std::ifstream::binary;
+  // asm2wasm.exe built with MinGW would crash with files containing '\r' when opened without the binary flag
```

Comment:
I think when we generate the asm.js text, on windows the text-handling code in windows just emits `\r`s automatically. @juj knows more.

If there is no danger here, then opening as binary sounds ok, but I don't know enough about windows and text handling there.